### PR TITLE
pull request for day 2

### PR DESCRIPTION
**Pull request by ryan4summer**
finish questions 209, 304, 53 and update the readme

### CODE REVIEW

File: tiantianXia/assignments/slidingwindow/lc209/lc209.js [hunk 1]
```diff
@@ -1,5 +1,45 @@
+// Problem Statement #
+
+// Given an array of positive numbers and a positive number ‘S,’ find the length of the smallest contiguous subarray whose sum is greater than or equal to ‘S’. Return 0 if no such subarray exists.
```
Comment (MelissaPastore) [inline tiantianXia/assignments/slidingwindow/lc209/lc209.js:3]: Nice pseudo code / summary!

File: tiantianXia/assignments/slidingwindow/lc340/lc340.js [hunk 2]
```diff
@@ -0,0 +1,114 @@
+// Problem Statement #
+
+// Given a string, find the length of the longest substring in it with no more than K distinct characters.
```
Comment (MelissaPastore) [inline tiantianXia/assignments/slidingwindow/lc340/lc340.js:3]: Again - nice summary / high level overview!

File: tiantianXia/assignments/slidingwindow/lc340/lc340.js [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+  return maxLen;
+}
+
+console.log(
+  `Length of the longest substring: ${longest_substring_with_k_distinct(
+    'araaci',
+    2
+  )}`
+);
+console.log(
```
Comment (MelissaPastore) [inline tiantianXia/assignments/slidingwindow/lc340/lc340.js:75]: Did you test your code before submitting it? It doesn't seem like it's passing all of these tests right now.

File: tiantianXia/assignments/slidingwindow/lc53/lc53.js [hunk 4]
```diff
@@ -1,5 +1,31 @@
+// Problem Statement #
+
+// Given an array of positive numbers and a positive number ‘k,’ find the maximum sum of any contiguous subarray of size ‘k’.
+
+/*
+inputs - an array of positive numbers, positive k for the subarray size
```
Comment (MelissaPastore) [inline tiantianXia/assignments/slidingwindow/lc53/lc53.js:6]: Could be good to write out a high level summary here.

**Review by MelissaPastore — CHANGES_REQUESTED** [review]
Please make sure to test your code before submitting - for lc340 - can you test your code and make sure it's passing the tests and then re-submit? 

Thanks!
Melissa

File: tiantianXia/assignments/slidingwindow/lc340/lc340.js (hunk 3 above)
Comment (ryan4summer) [reply tiantianXia/assignments/slidingwindow/lc340/lc340.js:75]: sorry, I misunderstood question. I thought the question is asking for no more than k same characters. I will change that.

**Review by ryan4summer — COMMENTED** [review]
> Please make sure to test your code before submitting - for lc340 - can you test your code and make sure it's passing the tests and then re-submit?
> 
> Thanks!
> Melissa

I will change the code and do another pull request

**Review by MelissaPastore — APPROVED** [review]

**Pull request merged**